homeassistant/components/camera/__init__.py

```diff
@@ -267,14 +256,10 @@ async def preload_stream(_):
             camera_prefs = prefs.get(camera.entity_id)
             if not camera_prefs.preload_stream:
                 continue
-
-            async with async_timeout.timeout(10):
-                source = await camera.stream_source()
-
-            if not source:
+            stream = await camera.create_stream()
+            if not stream:
                 continue
-
-            request_stream(hass, source, keepalive=True, options=camera.stream_options)
+            stream.stream_view_url("hls")
```

Comment:
Isn't it weird that requesting a url has the side effect of keeping the stream alive? Should we add a new method or constructor parameter to indicate `keepalive`?

Reply:
Actually, I see that it is already taken care off in `create_stream`. Is this call still necessary?

Reply:
I think the work happening is something like:
- `camera.create_stream` creates the Stream
- update options on the stream
- `stream.add_provider('hls')` to create the hls output and set an access token
- `stream.start` the background thread in case it died

All of this used to happen in `request_stream` so the challenge is to figure out the right division of labor into a usable API.  I was thinking the caller really is just saying: 
- Give me a stream,
- start a recording
- start an hls stream for viewing

An option here could be to add a function in camera that updates keepalive and returns the view url (I still like removing the duplicate prefs lookup).  I'll think about this more...

Reply:
My takeaway here is that that this was a bit too opaque.  I've unrolled some of this logic out of the `Stream` into `camera`, which I think may make the interaction a bit more clear with fewer side effects.